### ISSUE TITLE
Routes with a period trigger a full page reload

### ISSUE THREAD

**Issue by andrico1234** [withastro/astro#5458]
### What version of `astro` are you using?

1.6.10

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

yarn

### What operating system are you using?

Mac

### Describe the Bug

Hi all, 

I'm currently migrating the [Open UI site](https://open-ui.org/) to use Astro under the hood. Overall the process has been fairly straight forward, but I'm observing some issues with routing.

When I navigate to a route where the pathname contains a period, Astro triggers a full page reload.

/blog/hello-world (no full reload)

/components/hello-world.preview (full reload)

Here's a video of the video in action.

https://user-images.githubusercontent.com/17087167/203432510-ddba8a33-54df-4f5a-bb6d-5d32cf1ff220.mov

For the first handful of routes, the page navigates as expected. When I start navigating to routes with a period in the pathname, I'm noticing that the page is reloaded, and the assets are refetched, which seems to be causing some flashing of unstyled images.

Interestingly, I tried to repro this (linked in the repo), but paths with and without a period are causing full page reloads.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-vwsrqg?file=src/pages/page.2.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by bluwy** [comment]
Full page load between navigations is expected as Astro works as a multi-page application, where each page is it's own HTML. I think the issue is that the GitHub icon is experiencing FOUC specifically for pages with period, and I can't seem to replicate it. Can you provide a repro with how the icon is loaded?

**Comment by andrico1234** [comment]
Hey @bluwy , thanks for the quick response, 

After doing some more digging, I believe this behaviour was due to the turbolinks integration.

Sadly, it seems that this plugin is no longer supported. Since I removed the integration, I'm observing this pop-in behaviour for web fonts, even though I'm using the suggested method of using fontsource and adding the import statements to the top of the layout components. 

I'll update the repo to take this into consideration.

I managed to resolve the icon issue by apply height and width to the container element. It's not perfect, but it's gotten the job done.
